src/freehand_selection.jl

```diff
@@ -1,61 +1,188 @@
+            #println("Moving")
+            push!(moving,true)
+            push!(building,false)
+            push!(modifying,false)
+            push!(diff, XY(btn.position.x-value(ctx.points)[1].x,
+                           btn.position.y-value(ctx.points)[1].y))
+        elseif (ispolygon(value(ctx.points)) &&
+                !isinside(Point(btn.position),Point.(value(ctx.points))))
+            #println("Resetting")
+            push!(building,true)
+            # resets ctx.points
             push!(ctx.points,[])
-            Reactive.run_till_now()
-        end
-        if !ispolygon(value(ctx.points))
-            next = btn.position
-            if length(value(ctx.points)) > 3 # and click is near start
-                next = value(ctx.points)[1]
+            push!(num_pts, 0)
+        elseif isempty(value(ctx.points)) # adds first point
+            #println("First point")
+            push!(building,true)
+            push!(ctx.points, [btn.position])
+            push!(num_pts,1)
+        elseif !ispolygon(value(ctx.points)) # adds to polygon
+            if (length(value(ctx.points)) > 3 &&
+                value(ctx.points)[1].x - 5 <= btn.position.x <=
+                value(ctx.points)[1].x + 5 && value(ctx.points)[1].y - 5
+                <= btn.position.y <= value(ctx.points)[1].y + 5)
+                # finishes polygon if click near start
+                #println("Finishing")
+                push!(ctx.points,
+                      push!(value(ctx.points)[1:end-1],value(ctx.points)[1]))
+                push!(num_pts, length(value(ctx.points)))
+            else # adds to polygon
+                #println("Adding")
+                push!(ctx.points,push!(value(ctx.points)[1:end-1],btn.position))
+                push!(num_pts, length(value(ctx.points)))
             end
-            push!(ctx.points, push!(value(ctx.points), next))
         end
+        nothing
+    end
+
+    # When building the polygon, makes working point
+    sigbuild = map(filterwhen(mouse_motion,dummybtn, c.mouse.motion)) do btn
+        #println(".")
+        if !isempty(value(ctx.points)) && !ispolygon(value(ctx.points))
+            push!(ctx.points,
+                  push!(value(ctx.points)[1:value(num_pts)], btn.position))
+        end
+        nothing
     end
 
-    sigmove = map(filterwhen(enabled,dummybtn, c.mouse.motion)) do btn
-        
+    # Moves polygon
+    sigmove = map(filterwhen(moving,dummybtn,c.mouse.motion)) do btn
+        # move polygon
+        push!(ctx.points, move_polygon_to(value(ctx.points),XY(btn.position.x-value(diff).x,btn.position.y-value(diff).y)))
+        nothing
+    end
+
+    # Modifies polygon by handle
+    sigmodify = map(filterwhen(modifying,dummybtn,c.mouse.motion)) do btn
+        # moves clicked point
+        if value(modhandle) != 1 && value(modhandle) != -1
+            # move a middle point
+            A = value(ctx.points)
+            splice!(A, value(modhandle), [btn.position])
```

Comment:
Maybe another way to say it is this: let's say `value(modhandle) == 7`. Are you trying to replace the 7th entry with `btn.position`?
```julia
# Demo using splice
julia> a = collect(1:9)
9-element Array{Int64,1}:
 1
 2
 3
 4
 5
 6
 7
 8
 9

julia> splice!(a, 7, 0)
7

julia> a
9-element Array{Int64,1}:                                                                                                                                                                                                                    
 1                                                                                                                                                                                                                                           
 2                                                                                                                                                                                                                                           
 3                                                                                                                                                                                                                                           
 4                                                                                                                                                                                                                                           
 5                                                                                                                                                                                                                                           
 6                                                                                                                                                                                                                                           
 0                                                                                                                                                                                                                                           
 8                                                                                                                                                                                                                                           
 9                                                                                                                                                                                                                                           

# Demo using array indexing                                                                                                                                                                                                                                             
julia> a = collect(1:9)
9-element Array{Int64,1}:                                                                                                                                                                                                                    
 1                                                                                                                                                                                                                                           
 2                                                                                                                                                                                                                                           
 3                                                                                                                                                                                                                                           
 4                                                                                                                                                                                                                                           
 5                                                                                                                                                                                                                                           
 6                                                                                                                                                                                                                                           
 7                                                                                                                                                                                                                                           
 8                                                                                                                                                                                                                                           
 9                                                                                                                                                                                                                                           
                                                                                                                                                                                                                                             
julia> a[7] = 0
0                                                                                                                                                                                                                                            
                                                                                                                                                                                                                                             
julia> a
9-element Array{Int64,1}:                                                                                                                                                                                                                    
 1                                                                                                                                                                                                                                           
 2                                                                                                                                                                                                                                           
 3                                                                                                                                                                                                                                           
 4                                                                                                                                                                                                                                           
 5                                                                                                                                                                                                                                           
 6                                                                                                                                                                                                                                           
 0                                                                                                                                                                                                                                           
 8                                                                                                                                                                                                                                           
 9
```
Not only is the second one clearer and easier, it's also much more efficient computationally.